runtime/src/qio/qio.c

```diff
@@ -1085,15 +1108,23 @@ qioerr open_flags_for_string(const char* s, int *flags_out)
 static
 const char* string_for_open_flags(int flags)
 {
-  if( flags == O_RDONLY ) return "r";
-  else if( flags == O_RDWR ) return "r+";
-  else if( flags == (O_WRONLY | O_CREAT | O_TRUNC) ) return "w";
-  else if( flags == (O_RDWR | O_CREAT | O_TRUNC) ) return "w+";
-  else if( flags == (O_WRONLY | O_CREAT | O_APPEND) ) return "a";
-  else if( flags == (O_RDWR | O_CREAT | O_APPEND) ) return "a+";
+  // Also add 'e' at the end to indicate to glibc that the
+  // file should be opened with close-on-exec enabled.
+  if( flags == O_RDONLY )
+    return "r" QIO_FOPEN_CLOEXEC;
```

Comment:
Should there be a "+" here to concatenate the strings?  I'm not entirely certain how this builds otherwise, it's a feature of C with which I am unfamiliar.